### PR TITLE
Allow users to directly set feature output column names and save during serialization

### PR DESCRIPTION
**Pull request by thehomebrewnerd**
### Allow users to directly set feature output column names and save during serialization

Adds a `set_feature_names` method to `FeatureBase` to directly allow for setting of feature column names to override the default name. Also updates serialization to save and restore these names.

### CODE REVIEW

File: featuretools/tests/computational_backend/test_calculate_feature_matrix.py [hunk 1]
```diff
@@ -2266,3 +2275,25 @@ def multi_cum_sum(x):
...
+    direct_feat = ft.Feature(pd_es["sessions"].ww["device_name"], "log")
+    trans_feat = ft.Feature(pd_es["log"].ww["value"], primitive=Negate)
+    multi_output_agg_feat = ft.Feature(
+        pd_es["log"].ww["product_id"],
+        parent_dataframe_name="customers",
+        primitive=NMostCommon(n=2),
+    )
+
+    direct_name = ["device_direct"]
+    direct_feat.set_feature_names(direct_name)
```
Comment (rwedge) [inline featuretools/tests/computational_backend/test_calculate_feature_matrix.py:2290]: for single output features, do we want `set_feature_names` and `rename` to both work?

**Review by rwedge — COMMENTED** [review]
Can we test that a feature stacked on a renamed column from a feature with multiple outputs uses the renamed name?

File: featuretools/tests/computational_backend/test_calculate_feature_matrix.py (hunk 1 above)
Comment (thehomebrewnerd) [reply featuretools/tests/computational_backend/test_calculate_feature_matrix.py:2290]: Not sure - are you suggesting that we just allow `set_feature_names` to work for multi-output features and renaming feature columns for single output should be done through `rename` as you can be done currently?

Comment (thehomebrewnerd) [reply featuretools/tests/computational_backend/test_calculate_feature_matrix.py:2290]: As I think about it I think I like the idea of having `set_feature_names` operate only on multi-output features - seems more consistent with the method name, and would reduce the scope of the change. We could add a check in the method to raise an error if `number_output_features == 1`.

@rwedge What do you think about that approach?

Comment (thehomebrewnerd) [reply featuretools/tests/computational_backend/test_calculate_feature_matrix.py]: @rwedge Sounds good. I'll make this change and add the additional test case you recommended.

**Comment by thehomebrewnerd** [comment]
> Can we test that a feature stacked on a renamed column from a feature with multiple outputs uses the renamed name?

@rwedge I added a test that checks the name of a feature created by stacking on a `FeatureOutputSlice` of a multi-output feature. Also updated tests to include a multi-output GroupbyTransformFeature.

File: featuretools/feature_base/feature_base.py [hunk 2]
```diff
@@ -107,19 +107,26 @@ def get_feature_names(self):
     def set_feature_names(self, names):
         """Set new values for the feature column names, overriding the default values.
         Number of names provided much match the number of output columns defined for
-        the feature.
+        the feature. Only works for features that have more than one output column. Use
+        ``Feature.rename`` for changing the column name for single output features.
```
Comment (rwedge) [inline featuretools/feature_base/feature_base.py:111]:
```suggestion
        ``Feature.rename`` to change the column name for single output features.
```

**Review by rwedge — APPROVED** [review]

**Pull request merged**